src/app/app.component.ts

```diff
@@ -42,13 +45,22 @@ export class AppComponent implements OnInit {
     private configService: ConfigService,
     private entityMapper: EntityMapperService,
     private routerService: RouterService,
-    private entityConfigService: EntityConfigService
+    private entityConfigService: EntityConfigService,
+    private sessionService: SessionService,
+    private activatedRoute: ActivatedRoute,
+    private router: Router
   ) {
     // If loading the config earlier (in a module constructor or through APP_INITIALIZER) a runtime error occurs.
     // The EntityMapperService needs the SessionServiceProvider which needs the AppConfig to be set up.
     // If the EntityMapperService is requested to early (through DI), the AppConfig is not ready yet.
     // TODO fix this with https://github.com/Aam-Digital/ndb-core/issues/595
     configService.loadConfig(entityMapper);
+    // Reload config once the database is synced
+    sessionService
+      .getSyncState()
+      .waitForChangeTo(SyncState.COMPLETED)
+      .then(() => configService.loadConfig(entityMapper))
+      .then(() => router.navigate([], { relativeTo: this.activatedRoute }));
```

Comment:
if this is necessary to apply changes done by our `RouterService, the navigate should probably happen there in the `reloadRouting`?